README.md

```diff
@@ -1 +1,74 @@
-# depub
+# depub: minimise visibility
+
+## Overview
+
+When working on medium or large sized Rust code bases, it can be hard to know
+whether the visibility of functions, structs, and so on are still at the
+minimum required. For example, sometimes functions that once needed to be `pub`
+now only need to be `pub(crate)`, `pub(super)`, or simply private.
+
+`depub` minimises the visibility of such items in files passed to it, using a
+user-specified command (e.g. `cargo check`) as an oracle to tell if its
+reduction of an item's visibility is valid or not. In essence, `depub` does a
+string search for `pub`, replaces it with `pub crate` and sees if a test
+command still succeeds. If it does, it keeps that visibility, otherwise it
+replaces with the original and tries the next item. Note that `depub` is
+inherently destructive: it overwrites files as it operates, so do not run it on
+source code that you do not want altered!
+
```

Comment:
Do we need some kind of warning stating that: in general you cannot know what `pub` stuff downstream crates may consume, so you should only really use this tool in "enclosed ecosystems".

Reply:
> in general you cannot know what pub stuff downstream crates may consume, so you should only really use this tool in "enclosed ecosystems".

I thought that was sort-of obvious in this bit:

```
`depub` minimises the visibility of such items in files passed to it, using a
user-specified command (e.g. `cargo check`) as an oracle to tell if its
reduction of an item's visibility is valid or not.
```

but maybe it isn't?

Reply:
Well, I knew what you meant, but since the implications of getting it wrong could be quite bad, I don't think it hurts to make it explicit.

Reply:
Fixed (hopefully) in https://github.com/softdevteam/depub/pull/1/commits/fc38494984fbdc13186cfd764cdcdb24840befe1.